processor/compile/image.js

```diff
@@ -7,10 +7,9 @@ module.exports = function ImageCompiler() {
   visitors.image = function compile(node, ...args) {
     if (node.data?.hProperties?.className === 'emoji') return node.title;
 
-    // Use magic block instead of markdown when there are certain defined properties we can't store in markdown
     const { align, className, width } = node.data?.hProperties || {};
-    const useMagicBlock = Boolean(width) || className?.length || Boolean(align);
-    if (useMagicBlock) return visitors.figure.call(this, node, ...args);
+    const complexImage = Boolean(width) || className?.length || Boolean(align);
+    if (complexImage) return <Image {...node.data.hProperties} />;
```

Comment:
This needs to return a string?

Reply:
ahh, so like a template literal or `renderToString`?

honestly i'm not quite sure what this step is doing, so i'm kinda winging it haha

Reply:
Just a template literal. This is when the editor saves back to markdown/mdx. The editor calls `serialize`/`toString` which converts the slate value to `mdast`. Then it passes that to `RMDX.mdx`, which converts the `mdast` to a plain text string using all the compilers.